### PR TITLE
Ft refactor register form and user avatar

### PR DESCRIPTION
**Pull request by khalifan-kfan**
PR Checklist

- [ ] - Latest `master` branch merged
- [ ] - PR title descriptive (can be used in release notes)

#### Summary of Changes (What does this PR do?)
- Refactor registration component to allow for photo upload among other missing details as per the design.
- It also enable the user to have a user avatar with the first later of their name as the capital on a rounded background.

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Jira Ticket(s) Number
- [SA-15]

## Screenshots/Videos
image
image
image

### CODE REVIEW

**Comment by FaithDaka** [comment]
@khalifan-kfan Why isn't this ready to hit production?

File: src/pages/Authentication/Register.jsx [hunk 1]
```diff
@@ -93,6 +177,28 @@ const Register = () => {
...
+            <img
+              alt=""
+              src={
+                imageUrl === ""
+                  ? ProfilePlaceholder
+                  : URL.createObjectURL(imageUrl)
+              }
+            />
+          </section>
+          <input
```
Comment (FaithDaka) [inline src/pages/Authentication/Register.jsx:192]: This is nice!
But the button should be styled to match our design style.

File: src/pages/HomePage/Avatar.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+    const h = hash % 360;
+    return `hsl(${h}, 30%, 80%)`;
+  };
+  return (
+    <>
+   {img?
+     <img src={img} />:
+      <h1  style={{ backgroundColor: nameStringToHslColor(name), color: "#000" }}>
+       {name.charAt(0).toUpperCase()}
```
Comment (FaithDaka) [inline src/pages/HomePage/Avatar.jsx:26]: Great work!

**Review by FaithDaka — CHANGES_REQUESTED** [review]

**Comment by khalifan-kfan** [comment]
> @khalifan-kfan Why isn't this ready to hit production?

wasn't sure, gess it is

**Review by FaithDaka — APPROVED** [review]

**Pull request merged**